### ISSUE TITLE
Interim fix for the nightly itemized schedule updates

### ISSUE THREAD

**Issue by ccostino** [18F/openFEC#2390]
@LindsayYoung and I were discussing the impact of the nightly updates routinely failing and putting us in a perpetual state of requiring partitioning of the itemized schedule A and B data, and we came to the realization that there is a solution for a stop-gap fix until we're able to address the root cause of the issue (detailed in issues #2309 and #2227; we have a meeting setup with FEC folks next week).

The idea here is that we check for the presence of an empty recordset from the view before we process it and populate it in the designated queue tables.  [PostgreSQL provides a way to check for this easily](https://www.postgresql.org/docs/current/static/plpgsql-statements.html#PLPGSQL-STATEMENTS-DIAGNOSTICS), so we'll implement a hot fix to address it and at least get our nightly updates cleaned up now.

There's an additional task that we can do as a separate hot fix to keep track of which `sub_id`s fall into this state in a separate table and retry them at another time (e.g., a day later).  They'd remain in this table until the record is successfully processed, and it'll give us a bit more of a window into which records are falling into this state so we can understand why better.  Is it an expected thing (e.g., they don't exist in the corresponding `disclosure.f_item_receipt_or_exp` table) or something else?  It'll provide us with information to go on.

To recap:

* [x] Fix the nightly updates of itemized data by checking for empty resultsets before processing (#2392)
* [x] Add support for keeping track of and retrying failed `sub_id`s (#2398)

**Comment by ccostino** [comment]
For the second piece of these, we anticipate the parts/tasks to be as follows:

* Create two tables to track `sub_ids` not found for schedule A and B respectively
* Add support to the trigger methods (via an `else` clause to the empty resultset check) to put a `sub_id` not found into the corresponding table - might have to track the operation as well
* Add an additional nightly update task to process the `sub_id`s not found the previous day to see if they're now available in the view

**Comment by noahmanger** [comment]
Can this be closed now that #2398 is in?

**Comment by ccostino** [comment]
Yes, I think so.  If something else comes up with this it's going to be a new issue and require new work.  Thanks, @noahmanger!
